src/Compilers/Core/Portable/Operations/IConversionExpression.cs

```diff
@@ -17,14 +17,36 @@ public interface IConversionExpression : IHasOperatorMethodExpression
         /// Value to be converted.
         /// </summary>
         IOperation Operand { get; }
+
+#pragma warning disable RS0010 // Avoid using cref tags with a prefix
         /// <summary>
-        /// Kind of conversion.
+        /// Gets the underlying common conversion information.
         /// </summary>
-        ConversionKind ConversionKind { get; }
+        /// <remarks>
+        /// If you need conversion information that is language specific, use either
+        /// <see cref="T:Microsoft.CodeAnalysis.CSharp.IConversionExpressionExtensions.GetConversion(IConversionExpression)"/> or
+        /// <see cref="T:Microsoft.CodeAnalysis.VisualBasic.GetConversion(IConversionExpression)"/>.
+        /// </remarks>
+#pragma warning restore RS0010 // Avoid using cref tags with a prefix
+        CommonConversion Conversion { get; }
         /// <summary>
         /// True if and only if the conversion is indicated explicity by a cast operation in the source code.
         /// </summary>
-        bool IsExplicit { get; }
+        bool IsExplicitInCode { get; }
+        /// <summary>
+        /// True if the conversion will fail with a <see cref="InvalidCastException"/> at runtime if the cast fails. This is false for C#'s
+        /// <code>as</code> operator and for VB's <code>TryCast</code> operator.
+        /// </summary>
+        bool ThrowsExceptionOnFailure { get; }
+        /// <summary>
+        /// True if the conversion can fail at runtime with an overflow exception. This corresponds to C# checked and unchecked blocks.
+        /// </summary>
+        bool IsChecked { get; }
+        /// <summary>
+        /// The language that defined this conversion. Possible values are <see cref="LanguageNames.CSharp"/> and
+        /// <see cref="LanguageNames.VisualBasic"/>.
+        /// </summary>
+        string LanguageName { get; }
```

Comment:
@cston I'm not sure. As I mentioned to @CyrusNajmabadi below, I'm ok with moving `LanguageName`. However, for the rest of the fields, you can't infer everything here from the VB or C# `Conversion` structs. For C#, you could infer `IsExplicitInCode`, but for VB you'll need the corresponding BoundNode for the specific conversion. As far as I can tell, the existing `Conversion` structs do not provide any info on whether it was a checked or TryCast at all, so we'll again need the corresponding BoundNode to fill that in. Given that, I think I'd prefer to leave them in the `IConversionExpression`.